sklearn/tests/test_impute.py

```diff
@@ -490,3 +493,602 @@ def test_mice_additive_matrix():
+    msg = "Some column(s) have more than {}% missing values".format(
+        imputer.col_max_missing * 100)
+    assert_raise_message(ValueError, msg, imputer.fit, X)
+
+    # Test with insufficient number of neighbors
+    X = np.array([
+        [1, 1, 1, 2, np.nan],
+        [2, 1, 2, 2, 3],
+        [3, 2, 3, 3, 8],
+        [6, 6, 2, 5, 13],
+    ])
+    msg = "There are only %d samples, but n_neighbors=%d." % \
+          (X.shape[0], imputer.n_neighbors)
+    assert_raise_message(ValueError, msg, imputer.fit, X)
+
+    # Test with inf present
+    X = np.array([
+        [np.inf, 1, 1, 2, np.nan],
+        [2, 1, 2, 2, 3],
+        [3, 2, 3, 3, 8],
+        [np.nan, 6, 0, 5, 13],
+        [np.nan, 7, 0, 7, 8],
+        [6, 6, 2, 5, 7],
+    ])
+    msg = "+/- inf values are not allowed."
+    assert_raise_message(ValueError, msg, KNNImputer().fit, X)
+
+    # Test with inf present in matrix passed in transform()
+    X = np.array([
+        [np.inf, 1, 1, 2, np.nan],
+        [2, 1, 2, 2, 3],
+        [3, 2, 3, 3, 8],
+        [np.nan, 6, 0, 5, 13],
+        [np.nan, 7, 0, 7, 8],
+        [6, 6, 2, 5, 7],
+    ])
+
+    X_fit = np.array([
+        [0, 1, 1, 2, np.nan],
+        [2, 1, 2, 2, 3],
+        [3, 2, 3, 3, 8],
+        [np.nan, 6, 0, 5, 13],
+        [np.nan, 7, 0, 7, 8],
+        [6, 6, 2, 5, 7],
+    ])
+    msg = "+/- inf values are not allowed in data to be transformed."
+    assert_raise_message(ValueError, msg, KNNImputer().fit(X_fit).transform, X)
+
+
+def test_knn_n_neighbors():
+
+    X = np.array([
+        [0,       0],
+        [np.nan,  2],
+        [4,       3],
+        [5,       np.nan],
+        [7,       7],
+        [np.nan,  8],
+        [14,      13]
+    ])
+    statistics_mean = np.nanmean(X, axis=0)
+
+    # Test with 1 neighbor
+    X_imputed_1NN = np.array([
+        [0,      0],
+        [4,      2],
+        [4,      3],
+        [5,      3],
+        [7,      7],
+        [7,      8],
+        [14,     13]
+    ])
+
+    n_neighbors = 1
+    imputer = KNNImputer(n_neighbors=n_neighbors)
+
+    assert_array_equal(imputer.fit_transform(X), X_imputed_1NN)
+    assert_array_equal(imputer.statistics_, statistics_mean)
+
+    # Test with 6 neighbors
+    X = np.array([
+        [0,      0],
+        [np.nan, 2],
+        [4,      3],
+        [5,      np.nan],
+        [7,      7],
+        [np.nan, 8],
+        [14,      13]
+    ])
+
+    X_imputed_6NN = np.array([
+        [0,      0],
+        [6,      2],
+        [4,      3],
+        [5,      5.5],
+        [7,      7],
+        [6,      8],
+        [14,     13]
+    ])
+
+    n_neighbors = 6
+    imputer = KNNImputer(n_neighbors=6)
+    imputer_plus1 = KNNImputer(n_neighbors=n_neighbors + 1)
+
+    assert_array_equal(imputer.fit_transform(X), X_imputed_6NN)
+    assert_array_equal(imputer.statistics_, statistics_mean)
+    assert_array_equal(imputer.fit_transform(X), imputer_plus1.fit(
+        X).transform(X))
+
+
+def test_weight_uniform():
```

Comment:
please include knn_imputer in the test name. (Though it is tempting to expand the module into a package and keep a separate test file, I think we'll avoid that at this point)